systemtest/src/main/java/io/strimzi/systemtest/metrics/MetricsCollector.java

```diff
@@ -208,21 +215,51 @@ private int getDefaultMetricsPortForComponent() {
     /**
      * Parse out specific metric from whole metrics file
      * @param pattern regex pattern for specific metric
-     * @param data all metrics data
      * @return list of parsed values
      */
-    public static ArrayList<Double> collectSpecificMetric(Pattern pattern, Map<String, String> data) {
+    public ArrayList<Double> collectSpecificMetric(Pattern pattern) {
         ArrayList<Double> values = new ArrayList<>();
 
-        for (Map.Entry<String, String> entry : data.entrySet()) {
-            Matcher t = pattern.matcher(entry.getValue());
-            if (t.find()) {
-                values.add(Double.parseDouble(t.group(1)));
+        if (collectedData != null && !collectedData.isEmpty()) {
+            for (Map.Entry<String, String> entry : collectedData.entrySet()) {
+                Matcher t = pattern.matcher(entry.getValue());
+                if (t.find()) {
+                    values.add(Double.parseDouble(t.group(1)));
+                }
             }
         }
+
         return values;
     }
 
+    /**
+     * Method checks already collected metrics data for Pattern containing desired metric
+     * @param pattern Pattern of metric which is desired
+     *
+     * @return ArrayList of values collected from the metrics
+     */
+    public ArrayList<Double> waitForSpecificMetricAndCollect(Pattern pattern) {
+        synchronized (LOCK) {
```

Comment:
You can use synchronized keyword in the defining method scope:
```java
public ArrayList<Double> waitForSpecificMetricAndCollect(Pattern pattern) 
```
--
```java
public synchronized ArrayList<Double> waitForSpecificMetricAndCollect(Pattern pattern) 
```

Reply:
Good point, thanks! :)